src/parachain/escrow.ts

```diff
@@ -148,61 +156,94 @@ export class DefaultEscrowAPI implements EscrowAPI {
             this.getStakedBalance(accountId),
             this.systemAPI.getCurrentBlockNumber()
         ]);
-        const definedAmountToLock =
-            (
-                amountToLock
-                || newMonetaryAmount(0, this.governanceCurrency as Currency<GovernanceUnit>)
-            ) as typeof userStake;
-        const newUserStake = userStake.add(definedAmountToLock);
+        // Note: the parachain uses the balance_at which combines the staked amount
+        // and staked time to calculate the share in the reward pool
+        // like `amountLocked * unlockHeight`
+        // https://github.com/interlay/interbtc/blob/1.7.3/crates/escrow/src/lib.rs#L525
+        // We need to differentiate four cases:
+        // 1. The user does not add any new stake/extend their lock time
+        // 2. The user extends only the locktime
+        // 3. The user increases only their stake
+        // 4. The user extends the locktime and increases the stake
+
+        // Case 1: no change
+        let newUserStake = userStake;
+        let newTotalStake = totalStake;
+        const newStakedBalance = stakedBalance;
+        // Case 2 - 4
+        if (amountToLock || unlockHeight) {
+            // Maybe update the staked balance and the endBlock
+            const monetaryAddedStakedBalance = newMonetaryAmount(
+                amountToLock ? amountToLock.toBig() : 0,
+                this.governanceCurrency as Currency<GovernanceUnit>
+            );
+            newStakedBalance.amount = stakedBalance.amount.add(monetaryAddedStakedBalance);
+            newStakedBalance.endBlock = unlockHeight ? unlockHeight : stakedBalance.endBlock;
+
+            let addedStake = new Big(0);
+            // Case 2: user only plans to extend the lock
+            if ((amountToLock === undefined || amountToLock.isZero()) && unlockHeight) {
+                // Note: unlockHeight - currentBlockNumber should not be <0
+                addedStake.add(unlockHeight - currentBlockNumber);
```

Comment:
Since `addedStake` is added to `newUserStake`, it would be wrong to use a block number in that addition. I'm not sure of the parachain calculation but based on your addition below the block number is a multiplier. So `newUserStake` has to be scaled by `(newUnlockHeight - currentBlockNumber) / (unlockHeight - currentBlockNumber)` - it's probably better to handle changes to `newUserStake` in each conditional block individually